src/Misc/fileTools.ts

```diff
@@ -348,6 +349,23 @@ export class FileTools {
         };
 
         const requestFile = () => {
+            // For a Base64 Data URL we can parse the results and respond immediately
+            if (Tools.IsBase64(url)) {
+                try {
+                    onSuccess(useArrayBuffer ? Tools.DecodeBase64(url) : Tools.DecodeBase64AsText(url));
+                } catch (error) {
+                    const message = error.message || "Failed to parse the Data URL";
+
+                    if (onError) {
+                        onError(new RequestFileError(message, new WebRequest()));
+                    } else {
+                        Logger.Error(message);
+                    }
+                }
+
+                return;
+            }

```

Comment:
This code probably belongs in LoadFile and not RequestFile. LoadFile, for example, can also handle `file:` protocol. It probably makes the most sense to add `data:` protocol there.

Reply:
Hmm, just noticed that this won't work because `SceneLoader` is calling requestFile directly. Hmm..

Reply:
I'm actually a bit confused on the purpose of LoadFile versus RequestFile. LoadFile will pass the request to RequestFile if the url doesn't start with "file:" which means with my current implementation calling `LoadFile("data:base64,...")` will actually work because it will fall through to calling RequestFile.

If I move the logic into LoadFile then I would also need to update the SceneLoader to call LoadFile instead of directly calling ReadFile. SceneLoader would then also need to know that data URLs should be treated different than standard URLs.

I kind of see it the other way around actually. I think LoadFile and RequestFile should be combined to be one function so the consumer doesn't need to know that a "file:" url should be handled differently than a "data:" or "https:" url.

Reply:
RequestFile has additional callbacks specific to http requests. This was originally just LoadFile. I split this to support http range requests in the glTF loader.

> If I move the logic into LoadFile then I would also need to update the SceneLoader to call LoadFile instead of directly calling ReadFile. SceneLoader would then also need to know that data URLs should be treated different than standard URLs.

I have to try think about this more and maybe try some things.

Reply:
@bghgary has now reworked the change so this functionality has been moved into LoadFile rather than RequestFile. With that change SceneLoader just uses LoadFile and doesn't directly call RequestFile anymore.